napari/utils/events/event_utils.py

```diff
@@ -32,21 +32,30 @@ def connect_setattr(emitter: Emitter, obj, attr: str):
     ref = weakref.ref(obj)
 
     def _cb(*value):
-        setattr(ref(), attr, value[0] if len(value) == 1 else value)
+        if (ob := ref()) is None:
+            emitter.disconnect(_cb)
+            return
+        setattr(ob, attr, value[0] if len(value) == 1 else value)
 
     emitter.connect(_cb)
     # There are scenarios where emitter is deleted before obj.
     # Also there is no option to create weakref to QT Signal
     # but even if keep reference to base object and signal name it is possible to meet
     # problem with C++ "wrapped C/C++ object has been deleted"
+
+    # In all this 3 functions this should be uncommented instead of if clause in _cb
+    # but it causes segmentation fault in tests
```

Comment:
```suggestion
    # In all of these 3 functions, this should be uncommented instead of using
    # the if clause in _cb but that causes a segmentation fault in tests
```

Reply:
Does this sound OK @Czaki ?